pkg/lifecycle/render/helm/template.go

```diff
@@ -347,3 +359,135 @@ func (f *LocalTemplater) writeStateHelmValuesTo(dest string, defaultValuesPath s
 
 	return nil
 }
+
+// validate each file to make sure that it conforms to the yaml spec
+// TODO replace this with an actual validation tool
+func (f *LocalTemplater) validateGeneratedFiles(
+	fs afero.Afero,
+	dir string,
+) error {
+	debug := level.Debug(log.With(f.Logger, "method", "validateGeneratedFiles"))
+
+	debug.Log("event", "readdir", "folder", dir)
+	files, err := fs.ReadDir(dir)
+	if err != nil {
+		debug.Log("event", "readdir.fail", "folder", dir)
+		return errors.Wrapf(err, "failed to read folder %s", dir)
+	}
+
+	for _, file := range files {
+		thisPath := filepath.Join(dir, file.Name())
+		if file.IsDir() {
+			err := f.validateGeneratedFiles(fs, thisPath)
+			if err != nil {
+				return err
+			}
+		} else {
+			err := fixFile(fs, thisPath, file.Mode())
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func fixFile(fs afero.Afero, thisPath string, mode os.FileMode) error {
+	contents, err := fs.ReadFile(thisPath)
+	if err != nil {
+		return errors.Wrapf(err, "failed to read file %s", thisPath)
+	}
+
+	scanner := bufio.NewScanner(bytes.NewReader(contents))
+
+	lines := []string{}
+	for scanner.Scan() {
+		lines = append(lines, scanner.Text())
+	}
+	if err := scanner.Err(); err != nil {
+		return errors.Wrapf(err, "failed to read lines from file %s", thisPath)
+	}
+
+	lines = fixLines(lines)
+
+	var outputFile bytes.Buffer
+	for idx, line := range lines {
+		if idx+1 != len(lines) || contents[len(contents)-1] == '\n' {
+			fmt.Fprintln(&outputFile, line)
+		} else {
+			// avoid adding trailing newlines
+			fmt.Fprintf(&outputFile, line)
+		}
+	}
+
+	err = fs.WriteFile(thisPath, outputFile.Bytes(), mode)
+	if err != nil {
+		return errors.Wrapf(err, "failed to write file %s after fixup", thisPath)
+	}
+
+	return nil
+}
+
+// applies all fixes to all lines provided
+func fixLines(lines []string) []string {
+	for idx, line := range lines {
+		if argsLineRegex.MatchString(line) {
+			// line has `args:` and nothing else but whitespace
+			if !checkIsChild(line, nextLine(idx, lines)) {
+				// next line is not a child, so args has no contents, add an empty array
+				lines[idx] = line + " []"
+			}
+		} else if envLineRegex.MatchString(line) {
+			// line has `env:` and nothing else but whitespace
+			if !checkIsChild(line, nextLine(idx, lines)) {
+				// next line is not a child, so env has no contents, add an empty object
+				lines[idx] = line + " {}"
```

Comment:
why object?

Reply:
The short answer is "because otherwise `kustomize` crashes"
`env:` requires `{}`, `args:` requires `[]`, and `value:` requires a literal (for which we use `""`)